### DISCUSSION TITLE
Use only `noise` OR `sigma` as a parameter in gp.marginal_likelihood()

### DISCUSSION THREAD

**Discussion by danhphan** [pymc-devs/pymc3#6077] [General]
Hi, in `pymc.gp.gp.py` file, several GP classes such as "[Marginal](https://github.com/pymc-devs/pymc/blob/a90c420b8b31fe4996f11d8ca9b3cc80abf4a97d/pymc/gp/gp.py#L415)" and "[MarginalApprox"](https://github.com/pymc-devs/pymc/blob/a90c420b8b31fe4996f11d8ca9b3cc80abf4a97d/pymc/gp/gp.py#L729) use `noise` as a parameter in their `marginal_likelihood()` function, while other GP classes like "[MarginalKron](https://github.com/pymc-devs/pymc/blob/a90c420b8b31fe4996f11d8ca9b3cc80abf4a97d/pymc/gp/gp.py#L1138)" use `sigma` as a parameter.

And also in docstrings, sometimes we use `noise=sigma`, sometimes we use `sigma=sigma`.

As the use of `noise` and `sigma` in the gp context is similar, it could be better to use ONLY `noise` OR `sigma` in these GP classes. This could make the APIs for `marginal_likelihood()` function a bit more consistent to users.

I would prefer only keep `noise` as a parameter, as it represents the meaning of `noise` part in the equation: `y = f(x) + noise`. But using only `sigma` is also fine.

Would love to have your opinions on this. Thanks!

**Comment by fonnesbeck** [comment]
I would tend to lean towards `sigma` only because it is used elsewhere in PyMC as a standard deviation parameter. If we ever change the PyMC conventions from Greek to descriptive parameter names, then `noise` would make sense here. That said, since GPs are a quasi-independent submodule that have unique attributes so having different parameter names is not totally unreasonable. Its a question of whether we want parameter names consistent across PyMC or just across the GP submodule.

**Comment by danhphan** [comment]
Hi, yes agree, it should be better to make it consistent for other modules in PyMC. If it is alright, I will open an issue in PyMC to change the `noise` parameter to `sigma`. It should be an easy change (if nobody takes the issue, I will do it soon). Thank you :)

**Comment by bwengals** [comment]
Great idea, I like sigma too.  And it'd be good to be consistent.  

One thing that should be better documented though, is I think where `noise` is used, you can use a scalar, implying white noise, or an actual `Covariance`, so any process could be considered noise.  Where `sigma` is used, it has to be a scalar.   I think that must have been the point of the different naming.  It's confusing though, and it'd be better to be consistent, but documented.

So for instance here's this line in `Marginal`: https://github.com/pymc-devs/pymc/blob/0b191adc54080b4b89948cd74e30142b3a65b546/pymc/gp/gp.py#L449

And then `MarginalSparse` it has to be a scalar.  The implementations depend on white noise.   I think for VFE you can do more than white noise, but would need to double check.

**Comment by danhphan** [comment]
Yes, agree, we definitely need to update the docs to make it clearer. Many thanks @bwengals :)
